homeassistant/components/device_tracker/unifi.py

```diff
@@ -26,13 +26,14 @@
 NOTIFICATION_ID = 'unifi_notification'
 NOTIFICATION_TITLE = 'Unifi Device Tracker Setup'
 
+# TODO: Change CONF_VERIFY_SSL to allow True, False, or Path
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
     vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
     vol.Optional(CONF_SITE_ID, default='default'): cv.string,
     vol.Required(CONF_PASSWORD): cv.string,
     vol.Required(CONF_USERNAME): cv.string,
     vol.Required(CONF_PORT, default=DEFAULT_PORT): cv.port,
-    vol.Optional(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): cv.boolean,
+    vol.Optional(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): cv.boolean
```

Comment:
Use `vol.Any(cv.boolean, cv.string)`.

Reply:
Perhaps use `vol.IsFile()` instead of `cv.string`.

Reply:
Hi Martin, thanks for the heads up on this! I've implemented isfile and mocked the tests to pass for "valid" files existing and failing for non-existing files. My question now is I'm trying to have it fail tests with the verify_ssl set to incorrect types such as int, but this is passing through the boolean validator. Should I only test for True, False, paths; or should I be testing incorrect types? If so how do I get boolean to be more rigid / explicit?

Reply:
Is it really necessary to have it more rigid? The validation should make sure that errors and configuration mistakes are avoided. It doesn't have to catch all cases, I think, where the impact is low. I'd test for the cases that should work and the ones you want to avoid.

Reply:
Fair points. I'll remove the type checking tests and push up. Thanks again for your help.

Reply:
The new tests below seem to work for me if this is set to vol.Any(cv.boolean, cv.string). That's still better than the current code and will fix the feature.

Reply:
Just pushed the fix that allows a file to be passed. I've removed the two tests you mentioned @sdague as they were failing to raise due to boolean accepting `None` and `5555` values in favour of file existence validation. Do you think I should change it to cv.string instead? Essentially which validation is greater value, file exists or valid bool | string?